packages/context/src/resolution-session.ts

```diff
@@ -293,12 +276,43 @@ export class ResolutionSession {
   toString() {
     return this.getResolutionPath();
   }
+
+  /**
+   * Describe the injection for debugging purpose
+   *
+   * @TODO remove in next major version
```

Comment:
Use `@deprecated`?

Reply:
Deprecation does not count as breaking. It's signaling an upcoming breaking change.

Reply:
Also, there is a clear need to draw a line between:
1) End-User APIs -- to compose components and configs into system and eventually run it
2) Extension Author APIs -- Exposes API at the extension point in a coherent way
3) Low-level API  -- Stable API used to do low-lever things in the core (considered public API and under semver)
4) Internal API  -- It's exported but not considered stable, may change at any moment and must not be used outside of monorepo

Your thoughts on that?

At the moment, exported === public, which is incorrect. For example, when a new API surface is being created 
(by splitting a module into two, for example), we at first would export everything that crosses this API surface, 
and work out actual public API of the new module afterwards. But each time we use export we risk it to be 
treated as public API.

Long story short: exported functions should not be automatically treated as public. Public API should be designed with a great care. To me, it seems it already got to the point where inconsistent public API design hinders the development, 
and at the same time prevents from reaching any notion of stability (Long Term Support).

Reply:
I apologize if I have been too direct in the last message. I think it's an important piece of information (I looked through all discussions where the consensus has not been reached and tried to figure out the root cause so it can be fixed)

Reply:
It's easier than It sounded. Check this thing out: https://api-extractor.com/
It's created by MS to handle this exact kind of problem. 
What do you think on that?

Reply:
My Internal API actually is further divided into: 

- https://api-extractor.com/pages/tsdoc/tag_alpha/
- https://api-extractor.com/pages/tsdoc/tag_beta/
- https://api-extractor.com/pages/tsdoc/tag_internal/

That's looks even better

Reply:
> At the moment, exported === public, which is incorrect

Unfortunately that's not the case with LoopBack 4; `@public` is not used in most of its APIs, hence, IMO, marking it as a non-breaking change would not be in the best interest of the framework users.

**However**, #7533 is a breaking change, and we could drop a second PR to remove it completely.